src/dfm/common/data_cleaning/dolma_taggers/language_scandi.py

```diff
@@ -0,0 +1,202 @@
+"""
+
+Filters.
+
+"""
+from collections.abc import Iterable
+
+import pycld2 as cld2
+import regex
+from anyascii import anyascii
+from dolma.core.data_types import DocResult, Document, Span, TextSlice
+from dolma.core.ft_tagger import BaseFastTextTagger, Prediction
+from dolma.core.registry import TaggerRegistry
+from dolma.core.taggers import BaseTagger
+from dolma.core.utils import split_paragraphs
+
+LANGS = {
+    "ENGLISH": "en",
+    "DANISH": "da",
+    "SWEDISH": "sv",
+    "NORWEGIAN": "no",
+    "ICELANDIC": "is",
+    "FAROESE": "fo",  # Note that FAROESE is not supported by cld2
+}
+
+
+@TaggerRegistry.add("cld2_scandi_doc")
+class Cld2LanguageFilterScandi(BaseTagger):
+    RE_BAD_CHARS = regex.compile(r"[\p{Cc}\p{Cs}]+")
+
+    def _sanitize_input(self, text: str) -> str:
+        return self.RE_BAD_CHARS.sub("", text)
+
+    def _to_ascii_input(self, text: str) -> str:
+        return anyascii(text)
+
+    def _identity_fn(self, text: str) -> str:
+        return text
+
+    def _predict_text(self, text: str) -> dict[str, float]:
+        is_reliable = False
+        details: Iterable[tuple[str, str, int, float]] = []
+        for fn in (self._identity_fn, self._to_ascii_input, self._sanitize_input):
```

Comment:
Is there no case where `self._to_ascii_input(self._sanitize_input(text))` would work, but where `self._to_ascii_input` would not? 

If this works though, fuck it! ;-)

Reply:
It was able to run through mC4 Danish without problems, so I think it works.